cli/azd/cmd/env.go

```diff
@@ -164,13 +173,57 @@ func newEnvSetAction(
 }
 
 func (e *envSetAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	key := e.args[0]
-	value := e.args[1]
-
+	// To track case conflicts
 	dotEnv := e.env.Dotenv()
-	warnKeyCaseConflicts(ctx, e.console, dotEnv, key)
+	keyValues := make(map[string]string)
+
+	// Handle file input if specified
+	if e.flags.file != "" {
+		if len(e.args) > 0 {
+			return nil, fmt.Errorf("cannot combine --file flag with key-value arguments")
+		}
+		filename := e.flags.file
+		file, err := os.Open(filename)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open file %s: %w", filename, err)
+		}
+		defer file.Close()
+
+		keyValues, err = godotenv.Parse(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse file %s: %w", filename, err)
+		}
+	} else if len(e.args) == 0 {
+		//nolint:lll
+		return nil, fmt.Errorf("no environment values provided. Use '<key> <value>', '<key>=<value>', or '--file <filepath>'")
+	} else if len(e.args) == 2 && !strings.Contains(e.args[0], "=") {
+		// Handle single key-value pair format: azd env set key value
+		key := e.args[0]
+		value := e.args[1]
+		keyValues[key] = value
+	} else {
+		// Handle key=value format: azd env set key=value [key2=value2 ...]
+		for _, arg := range e.args {
+			key, value, err := parseKeyValue(arg)
+			if err != nil {
+				return nil, err
+			}
+			keyValues[key] = value
+		}
+	}
```

Comment:
We would also support multiple <key> <value> syntax like `azd env set key1 value1 key2 value2` or would we require `=` for setting multiple values like `azd env set key1=value1 key2=value2` ?

Reply:
Just the `azd env set key1=value1 key2=value2` syntax for setting multiple values